poetry/core/masonry/builders/builder.py

```diff
@@ -161,22 +161,20 @@ def find_files_to_add(
                                 to_add.add(include_file)
                     continue
 
-                if (
-                    isinstance(include, PackageInclude)
-                    and include.source
-                    and self.format == "wheel"
-                ):
-                    source_root = include.base
-                else:
-                    source_root = self._path
-
-                include_file = BuildIncludeFile(path=file, source_root=source_root)
+                include_file = BuildIncludeFile(path=file, source_root=self._path)
 
                 if self.is_excluded(
                     include_file.relative_to_source_root()
                 ) and isinstance(include, PackageInclude):
                     continue
 
+                if (
```

Comment:
Would a better fix here be to fix `is_excluded` to support `BuildIncludeFile` and detect that it is not in the project root? I just feel uncomfortable that we are updating the include file instance property. We should really make that immutable.

Reply:
Passing `BuildIncludeFile` to `is_excluded` doesn't work here, because in other parts of the code we need to pass a `Path`. But the solutions seems to be simpler. The `Builder` knows the path to the project root, so we can just drop it in here. For readability reasons I've added the project root as a new attribute to `BuildIncludeFile` and added a method `relative_to_project_root`, which we can use here.